Exceptions/ExceptionHandlingMiddleware.cs

```diff
@@ -0,0 +1,72 @@
+﻿using FluentValidation;
+using FluentValidation.Results;
+using System.Net;
+using System.Text.Json;
+
+namespace E_Learning.ErrorHandling
+{
+    public class ExceptionHandlingMiddleware
+    {
+        private readonly RequestDelegate _next;
+
+        public ExceptionHandlingMiddleware(RequestDelegate next)
+        {
+            _next = next;
+        }
+
+        public async Task Invoke(HttpContext context)
+        {
+            try
+            {
+                await _next(context);
+            }
+            catch (ValidationException ex)
+            {
+                await HandleValidationExceptionAsync(context, ex);
+            }
+            catch (Exception ex)
+            {
+                await HandleExceptionAsync(context, ex);
+            }
+        }
+
+        private Task HandleValidationExceptionAsync(HttpContext context, ValidationException ex)
+        {
+            context.Response.Clear();
+            context.Response.StatusCode = SetStatusCode(ex.Errors.FirstOrDefault());
+            context.Response.ContentType = "application/json";
+
+            var errors = ex.Errors.Select(e => new
+            {
+                Field = e.PropertyName,
+                Error = e.ErrorMessage
+            });
+
+            var result = JsonSerializer.Serialize(new { Errors = errors });
+            return context.Response.WriteAsync(result);
+        }
+
+        private int SetStatusCode(ValidationFailure validation)
+        {
+            if (validation.ErrorCode.Contains("not found"))
```

Comment:
Relying on string comparison is kind of error prone. A better approach would be to use enums, if you want to have a convention based implementation.
If you want to go with strings, at least use case invariants otherwise you might have issues with different cases in error code and your hardcoded string